src/org/way2it/data_structures/list/List.java

```diff
@@ -14,30 +16,78 @@ public List() {
     // Should add new value to the end of the list, increment length and extend array by x2 if needed
     public void add(String value) {
         // TODO implement me
+        valuesInc();
+        values[length]=value;
+        length++;
     }
 
     // Should add new value at the specified index, moving other values to the right,
     // increment length and extend array by x2 if needed
     public void addAtIndex(String value, int index) {
         // TODO implement me
+        valuesInc();
+
+        String[] temp = values;
+        values = new String[temp.length];
```

Comment:
Copying whole array to add one new element may cost much memory. Try to start iterating from the end of the array and move elements right till you find a position to insert a new value.